### PR TITLE
fix: refactor from ConfigFileFinder to load_config

### PR DESCRIPTION
**Pull request by lsorber**
Attempt at fixing https://github.com/tylerwince/flake8-bandit/issues/33 by refactoring from flake8<5.0.0's `ConfigFileFinder` to flake8>=5.0.0,<6.0.0's `load_config`.

### CODE REVIEW

**Comment by calumy** [comment]
This change now requires a `.bandit` file to be added; otherwise, this error is raised `Unable to parse config file: The specified config file does not exist: .bandit`. The [Bandit docs](https://bandit.readthedocs.io/en/latest/config.html#bandit-settings) suggest that the `.bandit` file is optional. 

Is this change intentional, or is there a way that this can be avoided? 

If this file is required, it might be worth updating the `README.md` to reflect this.

**Comment by lsorber** [comment]
@CalumY Apologies for the misleading output on stderr. A `.bandit` config file was and still is optional after this PR.

The message you're referring to was originally intended for when the `[bandit]` section of the `.bandit` file is missing, but this PR would also (erroneously) cause that message to be printed if the `.bandit` file itself is missing. This is entirely harmless. I've just pushed a commit that removes the message. Now, if the `.bandit` file is missing or not correctly formatted, its contents (if any) will silently be ignored.

**Comment by calumy** [comment]
@lsorber Thanks for sorting this!

**Comment by calumy** [comment]
Another possible way of sorting this would be to update the if statement to ignore the file not found error along with the unable to parse config file error. This would still allow an error when the `[bandit]` section of the `.bandit` file is missing as was originally intended but ignore the error if the `.bandit` file is missing altogether. 

```
if str(e) != "No section: 'bandit'" and str(e) != "The specified config file does not exist: .bandit":
    sys.stderr.write(f"Unable to parse config file: {e}")
```

**Comment by lsorber** [comment]
@CalumY Sure, that could be done, but OTOH I'm not sure it's this plugin's responsibility to verify the validity of the `.bandit` file. For example, the `[bandit]` section could exist but one of its settings may be misspelled. I'll leave it up to @tylerwince to decide on what to do with this.

File: setup.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def get_version(fname="flake8_bandit.py"):
 VERSION = get_version()
 
 # What packages are required for this module to be executed?
-REQUIRED = ["flake8", "bandit>=1.7.3", "flake8-polyfill", "pycodestyle"]
+REQUIRED = ["flake8>=5.0.0", "bandit>=1.7.3", "flake8-polyfill", "pycodestyle"]
```
Comment (Zac-HD) [inline setup.py]: FWIW neither flake8-polyfill nor pycodestyle are used anywhere in this repo; maybe we could remove the dependency at the same time?

Comment (lsorber) [reply setup.py]: Done!

**Comment by davfsa** [comment]
@tylerwince would it be possible to merge this and make a new release?

**Pull request merged**